hedera-node/src/main/java/com/hedera/services/txns/ethereum/EthereumTransitionLogic.java

```diff
@@ -0,0 +1,231 @@
+			AliasManager aliasManager,
+			final TransactionalLedger<AccountID, AccountProperty, MerkleAccount> accountsLedger) {
+		this.txnCtx = txnCtx;
+		this.spanMapAccessor = spanMapAccessor;
+		this.contractCallTransitionLogic = contractCallTransitionLogic;
+		this.contractCreateTransitionLogic = contractCreateTransitionLogic;
+		this.recordService = recordService;
+		this.hfs = hfs;
+		this.chainId = Integers.toBytes(globalDynamicProperties.getChainId());
+		this.aliasManager = aliasManager;
+		this.accountsLedger = accountsLedger;
+	}
+
+	@Override
+	public void doStateTransition() {
+		var syntheticTxBody = getOrCreateTransactionBody(txnCtx.accessor());
+		EthTxData ethTxData = spanMapAccessor.getEthTxDataMeta(txnCtx.accessor());
+		maybeUpdateCallData(txnCtx.accessor(), ethTxData, txnCtx.accessor().getTxn().getEthereumTransaction());
+		var ethTxSigs = getOrCreateEthSigs(txnCtx.accessor(), ethTxData);
+
+		var callingAccount = aliasManager.lookupIdBy(ByteString.copyFrom(ethTxSigs.address()));
+
+		if (syntheticTxBody.hasContractCall()) {
+			contractCallTransitionLogic.doStateTransitionOperation(syntheticTxBody, callingAccount.toId(), true);
+		} else if (syntheticTxBody.hasContractCreateInstance()) {
+			contractCreateTransitionLogic.doStateTransitionOperation(syntheticTxBody, callingAccount.toId(), true);
+		}
+		recordService.updateFromEvmCallContext(ethTxData);
+	}
+
+	private TransactionBody getOrCreateTransactionBody(final TxnAccessor txnCtx) {
+		var txBody = spanMapAccessor.getEthTxBodyMeta(txnCtx);
+		if (txBody == null) {
+			txBody = createSyntheticTransactionBody(spanMapAccessor.getEthTxDataMeta(txnCtx));
+			spanMapAccessor.setEthTxBodyMeta(txnCtx, txBody);
+		}
+		return txBody;
+	}
+
+	private EthTxSigs getOrCreateEthSigs(final TxnAccessor txnCtx, EthTxData ethTxData) {
+		var ethTxSigs = spanMapAccessor.getEthTxSigsMeta(txnCtx);
+		if (ethTxSigs == null) {
+			ethTxSigs = EthTxSigs.extractSignatures(ethTxData);
+			spanMapAccessor.setEthTxSigsMeta(txnCtx, ethTxSigs);
+		}
+		return ethTxSigs;
+	}
+
+	@Override
+	public Predicate<TransactionBody> applicability() {
+		return TransactionBody::hasEthereumTransaction;
+	}
+
+	@Override
+	public ResponseCodeEnum validateSemantics(TxnAccessor accessor) {
+		var ethTxData = spanMapAccessor.getEthTxDataMeta(accessor);
+		var txBody = getOrCreateTransactionBody(accessor);
+
+		if (ethTxData.chainId().length == 0 || Arrays.compare(chainId, ethTxData.chainId()) != 0) {
+			return ResponseCodeEnum.WRONG_CHAIN_ID;
+		}
+
+		if (accessor.getExpandedSigStatus() == ResponseCodeEnum.OK) {
+			// this is not precheck, so do more involved checks
+			maybeUpdateCallData(accessor, ethTxData, accessor.getTxn().getEthereumTransaction());
+			var ethTxSigs = getOrCreateEthSigs(txnCtx.accessor(), ethTxData);
+			var callingAccount = aliasManager.lookupIdBy(ByteString.copyFrom(ethTxSigs.address()));
+			if (callingAccount == null) {
+				return ResponseCodeEnum.INVALID_ACCOUNT_ID; 
+			}
```

Comment:
Looking at the [return value of `aliasManager.lookupIdBy()`](https://github.com/hashgraph/hedera-services/blob/09d41f87924346d09653e6ef9cda10e468e13b2d/hedera-node/src/main/java/com/hedera/services/ledger/accounts/AliasManager.java#L166-L168), should we check against `MISSING_NUM` and not `null`?

Reply:
Fixed.